nova/compute/manager.py

```diff
@@ -2629,17 +2629,15 @@ def _build_resources(self, context, instance, requested_networks,
             self.driver.failed_spawn_cleanup(instance)
             raise exception.BuildAbortException(instance_uuid=instance.uuid,
                     reason=e.format_message())
-        except Exception:
+        except Exception as e:
             LOG.exception('Failure prepping block device',
                           instance=instance)
             # Make sure the async call finishes
             if network_info is not None:
                 network_info.wait(do_raise=False)
                 self.driver.clean_networks_preparation(instance, network_info)
             self.driver.failed_spawn_cleanup(instance)
-            msg = _('Failure prepping block device.')
-            raise exception.BuildAbortException(instance_uuid=instance.uuid,
-                    reason=msg)
+            raise exception.BuildAbortException('Failure prepping block device. ' + str(e))
```

Comment:
We should still include the instance.uuid here, it is useful when looking through central logs to filter on instance UUID

Reply:
The max length of the error message is 225 defined in "instance_faults" tables. including instance_uuid takes a lot of space and will cut the message.

Reply:
Oh, that is quite annoying. `str(e)` doesn't print the whole trace does it? Just the summary? I'm surprised that Nova doesn't filter the instance_uuid part out of the message, given that it's already stored in a table w/ the instance UUID, and they control the exception handler.

Reply:
it won't print the stacktrace. just the message. event though, some messages are very long.